protobuf/echo/ServiceForwarder.cpp

```diff
@@ -74,10 +74,10 @@ namespace services
             processedSize += range.size();
         }
 
-        if (processedSize == forwardingSize || (contentsReader != nullptr && contentsReader->Empty()))
-            contentsReader = nullptr;
         if (processedSize == forwardingSize || (contentsWriter != nullptr && contentsWriter->Empty()))
             contentsWriter = nullptr;
+        if (processedSize == forwardingSize || (contentsReader != nullptr && contentsReader->Empty()))
```

Comment:
Could you add InSequence test for this? It seems pretty important to destroy contentsWriter first :)

Reply:
I'll think about the best  way to test this. For now, I'd like to move forward with this PR.